lib/phoenix/test/conn_test.ex

```diff
@@ -446,4 +446,76 @@ defmodule Phoenix.ConnTest do
       recycle(conn)
     end
   end
+
+  @doc """
+  Calls the `@endpoint` and invokes router pipeline while bypassing route match.
+  """
+  defmacro bypass(conn) do
+    quote bind_quoted: [conn: conn] do
+      Phoenix.ConnTest.bypass_pipeline(conn, @endpoint)
+    end
+  end
+  defmacro bypass(conn, router) do
+    quote bind_quoted: [conn: conn, router: router] do
+      Phoenix.ConnTest.bypass_pipeline(conn, @endpoint, router)
+    end
+  end
+  defmacro bypass(conn, router, pipeline) do
+    quote bind_quoted: [conn: conn, router: router, pipeline: pipeline] do
+      Phoenix.ConnTest.bypass_pipeline(conn, @endpoint, router, pipeline)
+    end
+  end
+
+  @doc """
+  Calls the Endpoint and bypasses Router match.
+
+  Useful for unit testing Plugs where Endpoint and/or
+  router pipline plugs are required for proper setup.
+
+  ## Examples
+
+  For example, Imagine you are testing an authentication
+  plug in isolation, but you need to invoke the Endpoint plugs
+  and `:browser` pipeline of your Router for session and flash
+  related dependencies:
+
+      conn =
+        conn
+        |> bypass(MyApp.Router, :browser)
+        |> MyApp.RequireAuthentication.call([])
+      assert conn.halted
+
+  Alternatively, you could inoke only the Endpoint, and Router:
+
+      conn =
+        conn
+        |> bypass(MyApp.Router)
+        |> MyApp.RequireAuthentication.call([])
+      assert conn.halted
+
+  Or only inoke the Endpoint's plugs:
+
+    conn =
+      conn
+      |> bypass()
+      |> MyApp.RequireAuthentication.call([])
+    assert conn.halted
+  """
+  @spec bypass_pipeline(Conn.t, Module.t) :: Conn.t
+  def bypass_pipeline(conn, endpoint),
```

Comment:
We have agreed we won't call the endpoint. You should pipe into `get/2` or whatever you want to yourself because get already has the convenience for recycling, setting http verb, params and so on. Ideally, those are the only signatures we need:

```elixir
defmacro bypass_pipeline(conn, router, pipelines \\ [])
def bypass_pipeline(conn, endpoint, router, pipelines)
```

Also, which name are we going with? `bypass`? `bypass_pipeline`? `bypass_through`?